### PR TITLE
chore: create PR for master sync after release

### PR DESCRIPTION
**Pull request by mertmit**
## Change Summary

Depends on #7940 - we can trigger a manual run after 7940 merged to test the action

- Action revised to sync master back to develop 
  - Create a PR instead of pushing (branch protection)
  - Target local SDK on automated PR

## Change type

- [x] chore: (updating grunt tasks etc; no production code change)

## Summary by CodeRabbit

- **Chores**
	- Updated the GitHub workflow for better synchronization between branches.
- **Refactor**
	- Enhanced the script for upgrading package versions with improved handling and environment-specific adjustments.

### CODE REVIEW

File: scripts/upgradeNocodbSdk.js [hunk 1]
```diff
@@ -5,6 +5,11 @@ const execSync = require('child_process').execSync;
 // extract latest version from package.json
 const nocodbSdkPackage = JSON.parse(fs.readFileSync(path.join(__dirname, '..', 'packages', 'nocodb-sdk', 'package.json'), 'utf8'))
 
+
+if (process.env.revertSDK === 'true') {
+  nocodbSdkPackage.version = 'workspace:^';
+}
+
```
Comment (pranavxc) [inline scripts/upgradeNocodbSdk.js:12]: we have to update package version in `nocodb` and `nc-gui` dependencies

Comment (pranavxc) [reply scripts/upgradeNocodbSdk.js:12]: version property in `nocodb-sdk/package.json` should remain same

Comment (mertmit) [reply scripts/upgradeNocodbSdk.js:12]: this script only updates nc-gui and nocodb package.json, nocodb-sdk version handled in bumpNocodbSdkVersion.js

Comment (pranavxc) [reply scripts/upgradeNocodbSdk.js:12]: Oh sorry, here it's just reading and updating value in object.

**Review by pranavxc — APPROVED** [review]

File: .github/workflows/sync-to-develop.yml [hunk 2]
```diff
@@ -9,15 +9,36 @@ jobs:
...
+        id: sync
         env:
-          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
\ No newline at end of file
+          GH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        run: |
+          cd nocodb
+          git fetch --quiet --unshallow origin
+          git checkout master
+          BRANCH_NAME=oss/$(git rev-list -n 3 HEAD | tail -1 | cut -c1-8)
```
Comment (wingkwong) [inline .github/workflows/sync-to-develop.yml:36]: suggest to add a comment on `git rev-list -n 3 HEAD | tail -1 | cut -c1-8`

**Review by wingkwong — APPROVED** [review]

File: .github/workflows/sync-to-develop.yml (hunk 2 above)
Comment (mertmit) [reply .github/workflows/sync-to-develop.yml]: I've added a comment and also changed the logic to use last commit hash which is simpler, thanks

**Pull request merged**